authentication/auth.go

```diff
@@ -0,0 +1,86 @@
+package authentication
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/skratchdot/open-golang/open"
+	"golang.org/x/oauth2"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+var (
+	conf      *oauth2.Config
+	ctx       context.Context
+	verifier  string
+	authToken string
+)
+
+func callbackHandler(w http.ResponseWriter, r *http.Request) {
+	queryParts, _ := url.ParseQuery(r.URL.RawQuery)
+
+	code := queryParts["code"][0]
+
+	tok, err := conf.Exchange(ctx, code, oauth2.VerifierOption(verifier))
+	if err != nil {
+		log.Fatal(err)
+	}
+	authToken = tok.AccessToken
+	_, err = io.WriteString(w, "Login successful! Please close this window and return back to CLI")
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func serve(done chan int) {
+	log.Fatal(http.ListenAndServe(":9998", nil))
```

Comment:
I'm not entirely sure what is happening here but it looks like we are relying on log.Fatal to close the connection? If so, let's try and properly close this without log.fatal

Reply:
httpListenAndServe returns an error, if one occurs, and log.Fatat is responsible for logging the error and closing one. I don't this we have any other approach for closing the connection as the documentation only specifies this one (Ref: https://pkg.go.dev/net/http#ListenAndServe)

Reply:
There exists a shutdown command https://pkg.go.dev/net/http#Server.Shutdown

Reply:
Example usage with some similar patterns https://stackoverflow.com/questions/39320025/how-to-stop-http-listenandserve

Reply:
Used http.Server to host and shutdown the handler